### PR TITLE
Budget endpoints

### PR DESCRIPTION
**Pull request by ChristosKehayas**
alahuakpar

### CODE REVIEW

File: src/SplitBackApi/Api/Endpoints/Authentication/AuthenticationEndpoints.cs [hunk 1]
```diff
@@ -8,14 +8,17 @@ public static partial class AuthenticationEndpoints {
       .WithTags("Authentication")
       .AllowAnonymous();
 
-    authGroup.MapPost("/verify-token", VerifyToken);
-    authGroup.MapPost("/sign-in", SignIn);
-    authGroup.MapPost("/request-sign-up", RequestSignUp);
-    authGroup.MapPost("/request-sign-in", RequestSignIn);
+    // authGroup.MapPost("/verify-token", VerifyToken);
```
Comment (thanoskat) [inline src/SplitBackApi/Api/Endpoints/Authentication/AuthenticationEndpoints.cs:11]: remove these

File: src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    IBudgetRepository budgetRepository,
+    IExpenseRepository expenseRepository,
+    ITransferRepository transferRepository,
+    IGroupRepository groupRepository,
+    ClaimsPrincipal claimsPrincipal,
+    HttpRequest request
+  )
+  {
+
+    var authenticatedUserId = "63ff33b09e4437f07d9d3982"; //claimsPrincipal.GetAuthenticatedUserId();
```
Comment (thanoskat) [inline src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs:25]: hardcoded

File: src/SplitBackApi/Data/Repositories/ExpenseRepository/ExpenseMongoDbRepository.cs [hunk 3]
```diff
@@ -26,86 +29,125 @@ IOptions<AppSettings> appSettings
...
+    var query = _expenseCollection.Find(filter);
+
+    int skipCount = (pageNumber - 1) * pageSize;
+    query = query.Limit(pageSize);
+    query = query.Skip(skipCount);
+
+    return await query.ToListAsync();
+  }
+
+  public async Task<Result<List<Expense>>> GetWhereMemberIsParticipant(string groupId, string memberId, DateTime startDate)
```
Comment (thanoskat) [inline src/SplitBackApi/Data/Repositories/ExpenseRepository/ExpenseMongoDbRepository.cs:115]: public async List<Expense> GetLatestByGroupIdMemberId

File: src/SplitBackApi/Data/Repositories/BudgetRepository/BudgetMongoDbRepository.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    var budgetFound = await _budgetCollection.Find(findBudgetFilter).FirstOrDefaultAsync();
+    if (budgetFound is null) return Result.Failure($"Budget from user {UserId} has not been found to be deleted");
+
+    await _budgetCollection.DeleteOneAsync(findBudgetFilter);
+    return Result.Success();
+
+  }
+
+  public async Task<Result<Budget>> GetByUserId(string UserId)
```
Comment (thanoskat) [inline src/SplitBackApi/Data/Repositories/BudgetRepository/BudgetMongoDbRepository.cs:45]: Task<Maybe<Budget>>

File: src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    var budgetTypeString = request.Query["budgetType"].ToString();
```
Comment (thanoskat) [inline src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs:26]: what if queries in request are missing

File: src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+      budgetCurrency = "USD";
+    }
+    else
+    {
+      day = budgetResult.Value.Day;
+      budgetType = budgetResult.Value.BudgetType;
+      budgetCurrency = budgetResult.Value.Currency;
+    }
+
+    var startDate = BudgetHelpers.StartDateBasedOnBudgetAndDay(budgetType, day).Value;
```
Comment (thanoskat) [inline src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs:54]: do not use Helper classes. Create a service class

Comment (ChristosKehayas) [reply src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs:54]: Done

Comment (ChristosKehayas) [reply src/SplitBackApi/Api/Endpoints/Budgets/BudgetEndpoint.BudgetInfo.cs:54]: ti zori travaei me ta statics to service?

**Pull request closed without merging**